src/screen/dashboard.rs

```diff
@@ -887,18 +902,39 @@ impl Dashboard {
                         }
                     }
                     ToggleNicklist => {
-                        if let Some((_, _, pane)) = self.get_focused_mut(main_window) {
-                            pane.update_settings(|settings| {
-                                settings.channel.nicklist.enabled =
-                                    !settings.channel.nicklist.enabled
+                        if let Some((_, pane, buffer_settings)) =
+                            self.get_focused_with_buffer_settings_mut(main_window)
+                        {
+                            let buffer = pane.buffer.data();
+
+                            let settings = pane.update_settings(|settings| {
+                                settings.channel.nicklist.toggle_visibility()
                             });
+
+                            if let Some(buffer) = buffer {
+                                buffer_settings.insert(buffer, settings);
+                            }
```

Comment:
You can see the issue I mentioned in the other comment here where we have 2 sources of truth. If we just store it at the buffer_settings level, we only have to update it in one place!